### DISCUSSION TITLE
Nested line comment in block comment -- VERY STRANGE

### DISCUSSION THREAD

**Discussion by guyutongxue** [typst/typst#1081] [General]
Seems that Typst supports nesting line comment in block comment, and it is [intentional](https://github.com/typst/typst/blob/bcc014c4e177cc4e8cf5ca8c24990908b507c0f8/tests/typ/compiler/comment.typ#LL20C1-L24C1).

But all other languages with C-style comments does **not** support this (C, C++, C#, Java, JavaScript, Rust, PHP, Swift, etc.).

This cause some strange error which is hard to discover:

```typst
Here is a sentence. /* We should add a reference from https://en.wikipedia.org/blahblah */ This is another sentence.

This is next paragraph.
```

In above example, all content after the "block comments" disappeared, because the "https://" is commenting "*/" out. Very strange, isn't it?

I suggest to change the syntax here, or specify this feature on the [documentation](https://typst.app/docs/reference/syntax/#comments).

**Comment by PgBiel** [comment]
Can you open an issue on this?

**Reply by guyutongxue** [reply]
OK, well... I do not think this is a "bug" or "feature request". Anyway, #1113 opened

**Reply by PgBiel** [reply]
Thanks. `//` commenting everything after the comment ends feels like a bug to me.
